build/Utils.cmake

```diff
@@ -151,3 +152,48 @@ function(extract_sources sources_file)
     endif()
   endif()
 endfunction()
+
+# Determine the path to buck2, taking in order of preference: explicitly
+# provided buck2 argument, system buck2 (if correct version), download/cache
+# buck2.
+function(resolve_buck2)
+  if(EXECUTORCH_ROOT)
+    set(executorch_root ${EXECUTORCH_ROOT})
+  else()
+    set(executorch_root ${CMAKE_CURRENT_SOURCE_DIR})
+  endif()
+
+  set(resolve_buck2_command
+    ${PYTHON_EXECUTABLE} ${executorch_root}/build/resolve_buck.py
+    --cache_dir=${CMAKE_CURRENT_BINARY_DIR})
+
+  if(NOT ${BUCK2} STREQUAL "")
+    list(APPEND resolve_buck2_command --buck2=${BUCK2})
+  endif()
+
+  execute_process(
+    COMMAND ${resolve_buck2_command}
+    OUTPUT_VARIABLE resolve_buck2_output
+    ERROR_VARIABLE resolve_buck2_error
+    RESULT_VARIABLE resolve_buck2_exit_code
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+    if(resolve_buck2_exit_code EQUAL 0)
+      message("${resolve_buck2_error}")
```

Comment:
Is this leftover debugging? If you mean to keep it, it should print to `message(STATUS`, but since this is the success case it doesn't seem necessary to print the error

Reply:
The logic here is that I wanted to log messages, such as indicating that buck2 was being downloaded, so that it ends up in the build log. I'm using stdout to return the resolved buck2 path, so I'm printing all logs from the script to stderr. There might be a better way to do this, but I have seen a number of cli tools log secondary output to stderr, so that's why I did it this way. I'm open to suggestions, though.